### PR TITLE
Better collateral address checking

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs [hunk 1]
```diff
@@ -26,26 +26,32 @@ public interface IJoinFederationRequestService
...
-            this.loggerFactory = loggerFactory;
+            this.logger = LogManager.GetCurrentClassLogger();
             this.network = network as PoANetwork;
             this.nodeSettings = nodeSettings;
             this.votingManager = votingManager;
         }
 
         public async Task<PubKey> JoinFederationAsync(JoinFederationRequestModel request, CancellationToken cancellationToken)
         {
+            // First ensure that this collateral address isnt already present in the federation.
```
Comment (quantumagi) [inline src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs:51]: A parallel request could already be in progress and this test would not catch that?

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs:51]: We could check the pending polls rather?

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs:51]: I think for now this is sufficient. Once we bring in your polls repository code we can check pending polls as it is possible that pending duplicate polls do exist.

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestService.cs:51]: We should be able to easily prevent multiple pending polls involving the same collateral address.

**Review by quantumagi — APPROVED** [review]
LGTM

**Review by zeptin — APPROVED** [review]

**Pull request merged**